widget/textgrid.go

```diff
@@ -310,6 +309,97 @@ func (t *TextGrid) refreshCell(row, col int) {
 	r.refreshCell(row, col)
 }
 
+// ForRangeFn calls back for each cell and row in the selection
+func (t *TextGrid) ForRangeFn(blockMode bool, startRow, startCol, endRow, endCol int, eachCell func(cell *TextGridCell), eachRow func(row *TextGridRow)) {
```

Comment:
regarding naming the `Fn` is not really needed, ForRange should be enough (as the function type is part of the signature). The documentation does not seem complete - `blockMode` is not explained.
Also the need for a row callback compared to cell is not clear - and the row callback is not called on first row - so it's more like a "eachNewRow" not just eachRow...?

You need to add the since like (blank comment line followed by a new comment // Since: 2.5" to mark this as a new API (same for the text getter added as well).

Reply:
Thank you, I will make these changes